src/platform/unix/mod.rs

```diff
@@ -415,18 +432,20 @@ impl OsIpcChannel {
 
 pub struct OsIpcReceiverSet {
     incrementor: Incrementor,
-    pollfds: Vec<pollfd>,
-    fdids: HashMap<c_int, u64, BuildHasherDefault<FnvHasher>>
+    poll: Poll,
+    pollfds: HashMap<Token, FDHandle, BuildHasherDefault<FnvHasher>>,
+    events: Events
 }
 
 impl Drop for OsIpcReceiverSet {
     fn drop(&mut self) {
-        unsafe {
-            for pollfd in self.pollfds.iter() {
-                let result = libc::close(pollfd.fd);
-                assert!(thread::panicking() || result == 0);
+        let mut result = 0;
+        for (_, &FDHandle{ id: _, descriptor: fd }) in &self.pollfds {
+            unsafe {
+                result |= libc::close(fd);
```

Comment:
While moving the `unsafe` block into the loop is probably a good thing (even if not strictly related to the actual change I think?), moving the `assert()` out of the loop seems like an unnecessary complication to me... Though admittedly that's rather nit-picky -- I guess that *might* just be a matter of taste?